### PR TITLE
Arf 97 support notifications api

### LINKED ISSUE #97: Support the Notifications API
https://developer.github.com/v3/activity/notifications/

Specifically I wanted to use github4s to [set a thread subscription](https://developer.github.com/v3/activity/notifications/#set-a-thread-subscription).

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #97 #119 Support Set a Thread Subscription from notification API

### CODE REVIEW

File: docs/src/main/tut/notification.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ - `ignored`: Determines if all notifications should be blocked from this thread
+
+```scala
+val threadSub = Github(accessToken).notifications.setThreadSub(5,true,false)
+threadSub.exec[cats.Id, HttpResponse[String]]() match {
+  case Left(e) => println("Something went wrong: s{e.getMessage}")
+  case Right(r) => println(r.result)
+}
+```
+See [the API doc](https://developer.github.com/v3/activity/notifications/#set-a-thread-subscription) for full reference.
```
Comment (fedefernandez) [inline docs/src/main/tut/notification.md]: Please, add this message to the end:
```
As you can see, a few features of the pull request endpoint are missing. As a result, if you'd like to see a feature supported, feel free to create an issue and/or a pull request!
```

File: docs/src/main/tut/notification.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+layout: docs
+title: Notification API
```
Comment (fedefernandez) [inline docs/src/main/tut/notification.md]: `Notifications API`

File: docs/src/main/tut/notification.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+---
+
+# Notification API
```
Comment (fedefernandez) [inline docs/src/main/tut/notification.md]: `Notifications API`

File: docs/src/main/tut/notification.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+Github4s supports the [Notification API](https://developer.github.com/v3/activity/notifications/). As a result,
```
Comment (fedefernandez) [inline docs/src/main/tut/notification.md]: `Notifications API`

File: github4s/shared/src/main/scala/github4s/api/Notifications.scala [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import github4s.GithubResponses.GHResponse
+import github4s.{GithubApiUrls, HttpClient, HttpRequestBuilderExtension}
+import github4s.free.domain._
+import github4s.free.interpreters.Capture
+import github4s.util.URLEncoder
+import io.circe.syntax._
+import io.circe.generic.auto._
+
+/** Factory to encapsulate calls related to Issues operations  */
```
Comment (fedefernandez) [inline github4s/shared/src/main/scala/github4s/api/Notifications.scala]: `Issues` ?

**Comment by fedefernandez** [comment]
giphy 2

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by AdrianRaFo** [comment]
@bond15 Sorry but this is spamming me a lot

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by bond15** [comment]
this is the comment

**Comment by 47degdev** [comment]
this is the comment

**Comment by 47degdev** [comment]
this is the comment

**Comment by 47degdev** [comment]
this is the comment

**Pull request merged**